src/snowflake/snowpark/functions.py

```diff
@@ -11544,3 +11545,52 @@ def randn(
         _emit_ast=False,
         _ast=ast,
     )
+
+
+@private_preview(version="1.29.0")
+@publicapi
+def ai_filter(
+    predicate: ColumnOrLiteralStr, expr: ColumnOrLiteralStr, _emit_ast: bool = True
+) -> Column:
+    """
+    Classifies free-form text into boolean based on a natural language predicate.
+    Returns a boolean value representing whether the predicate is valid for the provided text.
+    ``ai_filter`` will return NULL if the text is NULL.
+
+    Args:
+        predicate: The natural language condition that determines the result of the text string.
+        expr: A string containing the text to be classified.
+
+    Example::
+
+        >>> df = session.create_dataframe(["Switzerland", "Korea", "Panama"], schema=["country"])
+        >>> df.select(
+        ...     ai_filter("Is the country in Asia?", col("country")).as_("asia"),
+        ...     ai_filter("Is the country in Europe?", col("country")).as_("europe"),
+        ...     ai_filter("Is the country in North America?", col("country")).as_("north_america"),
+        ...     ai_filter("Is the country in Central America?", col("country")).as_("central_america"),
+        ... ).show()
+        -----------------------------------------------------------
+        |"ASIA"  |"EUROPE"  |"NORTH_AMERICA"  |"CENTRAL_AMERICA"  |
+        -----------------------------------------------------------
+        |False   |True      |False            |False              |
+        |True    |False     |False            |False              |
+        |False   |False     |False            |True               |
+        -----------------------------------------------------------
+        <BLANKLINE>
+        >>> df.filter(ai_filter("Is the country in Asia?", col("country"))).show()
+        -------------
+        |"COUNTRY"  |
+        -------------
+        |Korea      |
+        -------------
+        <BLANKLINE>
+    """
+    ast = build_function_expr("ai_filter", [predicate, expr]) if _emit_ast else None
```

Comment:
AST generation looks good, could you also please add a line in the `tests/ast/data/functions2.test` file using this method and update the expectation tests? (The added line should go under the `## TEST CASE` header, right above the `## EXPECTED UNPARSER OUTPUT` with a new numbered `df`.